### PR TITLE
Update radar_check.py

### PR DESCRIPTION
**Pull request by jmcouffin**
reviewed tayO radar check

### CODE REVIEW

**Comment by jmcouffin** [comment]
@tay0thman I reviewed more seriously your code, did quite a few edit for clarity and pep8 compliance
renamed lots of variables and methods.

- you need to review one last thing: pbt and psp aren't showing in display units
-  +nice to have, make X Y Z andangle when necessary in separate columns ;p

ref: #2418

File: extensions/pyRevitTools.extension/checks/radar_check.py [hunk 1]
```diff
@@ -3,144 +3,125 @@
...
-    def get_tempbbox(self, ToggleCAD, ToggleRVT, ToggleIFC, ToggleAll):
-        violatingCAD = []
-        violatingRVT = []
-        badelements = []
+    def get_tempbbox(self, toggle_CAD, toggle_RVT, toggle_IFC, toggle_all):
+        violating_CAD = []
+        violating_RVT = []
+        bad_elements = []
         # Create a 3D view
-        t = Transaction(doc, "Create 3D View")
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/radar_check.py:37]: I recommend using  `with` statement for any transaction, if is possible better use `revit.Transaction` [class](https://docs.pyrevitlabs.io/reference/pyrevit/revit/db/transaction/)

File: extensions/pyRevitTools.extension/checks/radar_check.py [hunk 2]
```diff
@@ -3,144 +3,125 @@
...
-#______________________________________________3D to Bounding Box Analysis
+
+# ______________________________________________3D to Bounding Box Analysis
 class Get3DViewBoundingBox():
-    def get_tempbbox(self, ToggleCAD, ToggleRVT, ToggleIFC, ToggleAll):
-        violatingCAD = []
-        violatingRVT = []
-        badelements = []
+    def get_tempbbox(self, toggle_CAD, toggle_RVT, toggle_IFC, toggle_all):
+        violating_CAD = []
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/radar_check.py:24]: Variable in function should be lowercase

File: extensions/pyRevitTools.extension/checks/radar_check.py [hunk 3]
```diff
@@ -3,144 +3,125 @@
...
+# ______________________________________________Global Variables
+doc = DOCS.doc  # type:Document
+INTERNAL_ORIGIN = (0, 0, 0)
+EXTENT_DISTANCE = 52800  # Linear Feet
 
-#______________________________________________3D to Bounding Box Analysis
+
+# ______________________________________________3D to Bounding Box Analysis
 class Get3DViewBoundingBox():
-    def get_tempbbox(self, ToggleCAD, ToggleRVT, ToggleIFC, ToggleAll):
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/radar_check.py:32]: Argument name should be lowercase

File: extensions/pyRevitTools.extension/checks/radar_check.py [hunk 4]
```diff
@@ -3,144 +3,125 @@
...
+        bad_elements = []
         # Create a 3D view
-        t = Transaction(doc, "Create 3D View")
+        t = DB.Transaction(doc, "Create 3D View")
         t.Start()
-        view3Dtypes = DB.FilteredElementCollector(doc).OfClass(DB.ViewFamilyType).WhereElementIsElementType().ToElements()
-        view3dtype = [v for v in view3Dtypes if v.ViewFamily == DB.ViewFamily.ThreeDimensional][0]
-        view = DB.View3D.CreateIsometric(doc, view3dtype.Id)
-        worksets = FilteredWorksetCollector(doc).OfKind(WorksetKind.UserWorkset).ToWorksets()
+        view_3D_types = DB.FilteredElementCollector(doc).OfClass(DB.ViewFamilyType).WhereElementIsElementType().ToElements()
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/radar_check.py:30]: PEP 8: E501 line too long (124 > 120 characters)

File: extensions/pyRevitTools.extension/checks/radar_check.py [hunk 5]
```diff
@@ -3,144 +3,125 @@
...
                                             forEditing=False)
-
-
     else:
-        UIunits = DB.UnitFormatUtils(units=doc.GetUnits(),
-                                     unitType=DB.UnitType.UT_Length,
-                                        value=distance,
-                                        maxAccuracy=False,
-                                        forEditing=False)
+        ui_units = DB.UnitFormatUtils.Format(units=doc.GetUnits(), unitType=DB.UnitType.UT_Length, value=distance, maxAccuracy=False, forEditing=False)
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/radar_check.py:99]: It is advisable to make the code formatting similar in the same places

File: extensions/pyRevitTools.extension/checks/radar_check.py [hunk 6]
```diff
@@ -355,18 +329,17 @@ class ModelChecker(PreflightTestCase):
         - The bounding box of the CAD and RVT links
         - The bounding box of all elements in the model
     """
-
     name = "10 Mile Radar"
     author = "Tay Othman"
 
     def setUp(self, doc, output):
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/radar_check.py:335]: Shadows name 'doc' from outer scope

File: extensions/pyRevitTools.extension/checks/radar_check.py [hunk 7]
```diff
@@ -266,20 +241,19 @@ def check_model_extents(doc, output):
                 print(output.linkify(y.Id)+ str(y.Name)+ " - Is part of design option - "+ str(y.DesignOption.Name) )
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/radar_check.py:241]: missing spaces, formatting is messed up

File: extensions/pyRevitTools.extension/checks/radar_check.py [hunk 8]
```diff
@@ -3,144 +3,125 @@
...
-                    bbox = el.get_BoundingBox(view)
-                    if analyzebbox(bbox, intOrig, 52800) == 0:
-                        badelements.append(el)
-
+        if toggle_all:
+            elements = DB.FilteredElementCollector(doc, view.Id).WhereElementIsNotElementType().ToElements()
+            for element in elements:
+                if element.get_BoundingBox(view) is not None and hasattr(element, 'Name') and hasattr(element, 'Category'):
+                    bbox = element.get_BoundingBox(view)
+                    if check_bounding_box(bbox, INTERNAL_ORIGIN, 52800) == 0:
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/radar_check.py:76]: magic number, needs create constant or comment with unit of measurement

File: extensions/pyRevitTools.extension/checks/radar_check.py [hunk 9]
```diff
@@ -149,94 +130,88 @@ def analyzebbox(bbox, intOrig, extentdistance):
...
+    # ______________________________________________HTML Styles
     output = script.get_output()
     output.add_style('cover {color:black; font-size:24pt; font-weight:bold;}')
     output.add_style('header {color:black; font-size:15pt;}')
-    stringseperator = "_________________________________________________________________________________________________"
-    TestScore = 0
-    #__________________________________________check the distnaces of base and survey points
+    divider = "_"*100
+    test_score = 0
+    # __________________________________________check the distnaces of base and survey points
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/radar_check.py:164]: misspelling `distnaces`

File: extensions/pyRevitTools.extension/checks/radar_check.py [hunk 10]
```diff
@@ -149,94 +130,88 @@ def analyzebbox(bbox, intOrig, extentdistance):
...
-                    ['Project Base Point Distance from Internal Origin', str(convert_units(doc, baseptdistance))],
-                    ['Survey Point Distance from Internal Origin', str(convert_units(doc, surveydistance))],
-                    ['Project Base Point to Survey Delta X', str(convert_units(doc, calculate_Horizontal_Distance(basept, survpt)[1]))],
-                    ['Project Base Point to Survey Delta Y', str(convert_units(doc, calculate_Horizontal_Distance(basept, survpt)[2]))],
-                    ['Horizontal Distance between Project Base Point and Survey Point', str(convert_units(doc, calculate_Horizontal_Distance(basept, survpt)[0]))],
-                    ['Project Elevation', str(convert_units(doc, (survpt[2] - basept[2])))]]
-
+        output.print_md('### :OK_hand_medium_skin_tone: ............Survey Point is less than 10 miles (16KM) away from the Internal Origin.')
+    baseptdistance = calculate_distance(basept, INTERNAL_ORIGIN)
+    tabledata = [['InternaL Origin Coordinates', str(INTERNAL_ORIGIN)],
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/radar_check.py:180]: misspelling `InternaL`

File: extensions/pyRevitTools.extension/checks/radar_check.py [hunk 11]
```diff
@@ -300,53 +274,53 @@ def check_model_extents(doc, output):
...
+        output.print_md('### :OK_hand_medium_skin_tone: ............All CAD and RVT Links are located less than 10 miles (16KM) away from the Internal Origin.')
+        test_score += 1
+        print(divider)
+    if check_bounding_box(cleanbbox, INTERNAL_ORIGIN, 5) == 0:
         output.print_md('### :thumbs_down_medium_skin_tone: ............Distant objects are still being detected!')
         output.print_md('### :warning: ............Further Analysis Required.')
     else:
-        output.print_md('### :OK_hand_medium_skin_tone: ............All Objects are located less than 10 miles (16KM) away from the Internal Origin.***')
+        output.print_md('### :OK_hand_medium_skin_tone: ............All Objects are located less than 10 miles (16KM) away from the Internal Origin.')
         sys.exit()
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/radar_check.py:285]: I think better to use `script.exit()` [docs](https://docs.pyrevitlabs.io/reference/pyrevit/script/#pyrevit.script.exit)

**Review by dosymep — CHANGES_REQUESTED** [review]
@tay0thman, you need to check 
 - code formatting, lots of missing and extra spaces, 
 - non-pep8 variable naming,
 - the length of the lines should not be more [around 79 symbols](https://peps.python.org/pep-0008/#maximum-line-length)
 - shadows names

**Comment by tay0thman** [comment]
Thank you @dosymep for your thorough review and feedback, good to know how important these items are and will change the code to reflect these recommendations very shortly.

Thanks again

**Review by tay0thman — APPROVED** [review]
These Edits makes sense, I am approving them and testing on my scenarios using multiple models. I'll carry all necessary edits to meet with the additional comments.

**Review by tay0thman — APPROVED** [review]

File: extensions/pyRevitTools.extension/checks/radar_check.py (hunk 1 above)
Comment (tay0thman) [reply extensions/pyRevitTools.extension/checks/radar_check.py:37]: I just learned the wrapped dry-transaction, pretty cool!

File: extensions/pyRevitTools.extension/checks/radar_check.py (hunk 2 above)
Comment (tay0thman) [reply extensions/pyRevitTools.extension/checks/radar_check.py:24]: done!

File: extensions/pyRevitTools.extension/checks/radar_check.py (hunk 3 above)
Comment (tay0thman) [reply extensions/pyRevitTools.extension/checks/radar_check.py:32]: done!

File: extensions/pyRevitTools.extension/checks/radar_check.py (hunk 4 above)
Comment (tay0thman) [reply extensions/pyRevitTools.extension/checks/radar_check.py:30]: I am updating the script to limit my lines to 80 characters. I am exceeding these for few lines but will make sure it is less than 120

Comment (dosymep) [reply extensions/pyRevitTools.extension/checks/radar_check.py:30]: I use [pyCharm](https://www.jetbrains.com/pycharm), it has a vertical line and hints about it, but I think in vs code there are also the same settings or plugins for tracking coding style

image

**Comment by tay0thman** [comment]
Hello @jmcouffin  and @dosymep  I carried the updates and generated a new pull request.

File: extensions/pyRevitTools.extension/checks/radar_check.py (hunk 4 above)
Comment (tay0thman) [reply extensions/pyRevitTools.extension/checks/radar_check.py:30]: yup, I was able to achieve the same with vscode :)

**Pull request closed without merging**